src/sinks/datadog/metrics/tests.rs

```diff
@@ -0,0 +1,153 @@
+#![allow(clippy::print_stdout)] // tests
+
+use super::DatadogMetricsConfig;
+use crate::{
+    config::SinkConfig,
+    sinks::util::test::{build_test_server_status, load_sink},
+    test_util::{map_event_batch_stream, next_addr},
+};
+use bytes::Bytes;
+use flate2::read::ZlibDecoder;
+use futures::{channel::mpsc::Receiver, stream, StreamExt};
+use hyper::StatusCode;
+use indoc::indoc;
+use vector_core::event::{BatchNotifier, BatchStatus, Event, Metric, MetricKind, MetricValue};
+
+// The sink must support v1 and v2 API endpoints which have different codes for
+// signaling status. This enum allows us to signal which API endpoint and what
+// kind of response we want our test to model without getting into the details
+// of exactly what that code is.
+enum ApiStatus {
+    OKv1,
+    // OKv2,
```

Comment:
Do we really have v2 for this endpoint? https://docs.datadoghq.com/api/latest/events/#get-an-event